element-hq/element-x-android
# Feature: identity reset

## Content

- Moves OIDC related code to their own modules: `libraries/oidc`.
- Improves the custom chrome tab implementation and fixes issues with the fallback webview one (it was freezing after signing in).
- Creates `OidcEntryPoint` which allows opening any of the OIDC flow options (custom tab, webview).
- Adds new flows for identity reset: root screen with confirmation, OIDC one which uses the extracted OIDC logic, user+password one.
- Create `IdentityResetHandle` wrappers for both OIDC and password variants.

## Motivation and context

Closes #3268.

## Screenshots / GIFs

In the PR.

## Tests

With a user + password based account whose data you don't mind losing:

- Log in.
- Select 'Can't confirm?'.
- Continue, add your password and confirm.
- You should be fully logged in now and the system will ask you to add key backup.

With an OIDC one (maybe using `synapse-oidc.element.dev`?):

- Log in.
- Select 'Can't confirm?'.
- Continue, confirm the reset in the custom tab/webview, then go back.
- You should be fully logged in now and the system will ask you to add key backup.

## Tested devices

- OS version(s): 13

## Checklist
